### PR TITLE
Use slice instead of map for ingress paths

### PR DESCRIPTION
**Pull request by orangedeng**
Related Issue: https://github.com/rancher/rancher/issues/17065

### CODE REVIEW

**Comment by orangedeng** [comment]
@cjellick Logic updated, API won't break in the current version. We just sort the paths in reverse order

**Comment by loganhz** [comment]
@gitlawr Please review it

**Comment by gitlawr** [comment]
It looks good, but I wonder if we can put this mapper to `norman/types/mapper` as it looks generic.
cc @cjellick @ibuildthecloud

**Comment by alena1108** [comment]
@cjellick there are no more breaking api changes, but still a decision to be made of where the sort method should go to - types or norman library

File: mapper/sort_map_slice.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"sort"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+)
+
+type SortMapSlice struct {
+	Field        string
+	SorterKey    string
+	ReverseOrder bool
```
Comment (cjellick) [inline mapper/sort_map_slice.go]: You only ever set reverOrder to true. Why over-complicate by adding this boolean?

**Comment by loganhz** [comment]
@cjellick Do you mean we need to make a breaking API change? Change it from `map` to `array`?

**Comment by orangedeng** [comment]
Updated.

**Comment by cjellick** [comment]
@orangedeng The change looks fine, but it definitely breaks the UI. Open a UI issue and get a UI PR submitted and reviewed so that we can simultaneously merge the UI and the backend.
cc @loganhz

**Comment by cjellick** [comment]
Waiting on UI to be done so I can coordinate merging, since this breaks the UI https://github.com/rancher/rancher/issues/17529

**Comment by loganhz** [comment]
UI PR is ready for review and merge https://github.com/rancher/ui/pull/2585

**Comment by cjellick** [comment]
LGTM

**Pull request merged**